### PR TITLE
Borgs are now stunned by AoE flash

### PR DESCRIPTION
**Pull request by PKPenguin321**
:cl:
balance: Borgs will now get stunned by AoE flashes
/:cl:

Why:
~~Requested~~ Demanded by the design lead

ss 2018-09-05 at 08 39 32

### CODE REVIEW

**Comment by zxaber** [comment]
I assume this is the lead-up to remove Malf for being a waste of antag rep?

**Comment by Poojawa** [comment]
Borgs are just useful tools, they're not essential to a malf's victory. just ensure that you've a borg printer or something.

**Comment by Vidiveni** [comment]
Does this work with the flash from flash powder after this PR?

**Comment by Tlaltecuhtli** [comment]
just dodge lmao

**Comment by Arkvold** [comment]
I'm honestly okay with this.  Flashes are one of the real major weaknesses of borgs in a head-to-head fight.  Being vulnerable to flashes is just one of the prices you have to pay to be a silicon.  You want flash immunity, you need to be a fleshy and wear eye protection.

Don't see a reason why a borg would be immune to AoE flashes, but not direct ones; overloaded optic sensors are overloaded optic sensors.  So long as they're able to use LoS and speed to escape flashbangs just like a fleshy can, I see no problems with it.

**Comment by zxaber** [comment]
>Don't see a reason why a borg would be immune to AoE flashes, but not direct ones; overloaded optic sensors are overloaded optic sensors.

You could ask the same question on why humans aren't flashed by AoE.

**Comment by alexkar598** [comment]
I say give all borgs a module to aoe stun all humans on range for 20 seconds

**Comment by BeeSting12** [comment]
👍 

Silicon players can cry all they want but this is a good change. @optimumtact's vision for the game is turning out really well.

**Comment by ExcessiveUseOfCobblestone** [comment]
Bring back the "silicons are a punishment role" design.

**Comment by zombiegums** [comment]
This change is bad and oranges should feel bad. He is no longer best fruit. :^(

On a more serious note, I cry. Borg sprites are already thicc and much easier to click (thus, easier to flash IMO) than organic sprites. Plus, they cannot wear sunglasses or any other form of eye protection which I think is already a big weakness (and a fine one to have). So, is this change going to affect organics as well, given that they actually have methods of becoming immune to flashes, while borgs do not? If not, will the AoE flash function be removed from borgs (not even sure why it exists at all since it currently does nothing outside of being a good way to prank people into thinking it's revs) given that I do not want to accidentally flash myself if this change does goes through.

If this is going to go through (please, let this be a ruse cruise), it needs to be consistent and affect everyone who does not have appropriate eye protection, not just borgs.

**Comment by ExcessiveUseOfCobblestone** [comment]
Borgs don't have glasses but they have stun protection from literally every other form of stun that affects carbons.

**Comment by alexkar598** [comment]
Traitors and sec have access to flashes via either looting officers,load out and robotics and clings/cult have both round start aoe stun for Borg in the form of emps. I main sec and Borg(I do avoid combat as Borg tho) and its already very easy to take out Borgs if your prepared(emp/flash)

**Comment by zombiegums** [comment]
> Borgs don't have glasses but they have stun protection from literally every other form of stun that affects carbons.

Outside of chemicals, stunprods/stunbatons and slips, what ones are those? Flashes, flashbangs, EMP's, laser-pointers, borg stun-arm, goliath tendrils, mech knockback, wiz missiles (I'm sure there's more), being tabled (glass) affect borgs. 

I dislike this suggestion, but with regard to the first 4 - the immunity makes sense **consistency-wise** because borgs are machines. The only one that's a bit odd is the stun-arm. The AoE flash only affecting borgs doesn't make sense with regard to consistency, given that both borgs and organics without eye protection are stunned by wall-mounted flashes.

**Comment by Supermichael777** [comment]
thistookwaytoolong

**Comment by PKPenguin321** [comment]
About flash powder: Not gonna touch it in this PR or at all until https://github.com/tgstation/tgstation/issues/40144 is addressed

**Comment by Wilchenx** [comment]
@PKPenguin321 if static flashers stun humans aoe as well as borgs, then why does small aoe flash not stun humans aswell?

**Comment by ExcessiveUseOfCobblestone** [comment]
Because flashers of all variants are dumb and need 40144. Flashbangs should equal Flashers should equal Flashes

**Comment by PKPenguin321** [comment]
yeah it's AoE

**Comment by zombiegums** [comment]
After seeing that range, if that goes through then RIP borgs actually being playable (and fun). >:^(
Given that when a borg is stunned by a flash, it cannot do anything, not even speak on any channel, including binary, (unlike organics unless flashed by a headrev) - this is even more awful. 

A dinky flash that costs literally nothing to produce and are usually a dime-a-dozen should not be that powerful. Absolute madlads.

**Comment by Potato-Masher** [comment]
Okay, now that's just not good at all. 

This puts an already insane counter to borgs on another level of existence and power that is usually reserved to wiznerds (exaggeration but you get the idea).

**Comment by PKPenguin321** [comment]
Sorry but you're just going to have to take it up with the design lead.

**Comment by Arianya** [comment]
I disagree with this change, not because of any borg sympathies, but because it's giving additional utility to the flash that was previously the realm of flashbangs/the flasher.

While ensuring that stuns for borgs remain easily accessible is important, lets not reduce the use cases for other, more specialized items by over generalizing.

**Comment by Bawhoppen** [comment]
It should probably affect other mobs too, since it doesn't affect humans in AoE either...

**Comment by kriskog** [comment]
Lots of people moaning about borgs having less combat ability with this change. Why should borgs have combat ability in the first place? They are helper robots, not war robots.

**Comment by Wilchenx** [comment]
>not war robots.

This change also makes syndieborgs even more trash, now just about any assistant can suicidally run towards a syndieborg and stun it to death, instead of actually aiming with laser pointer or ion gun.

**Comment by kriskog** [comment]
If that is a problem, make changes to the actual war robots to be more resistant.

**Comment by zxaber** [comment]
Borgs have a duty (Law 1) to try and prevent harm. This would include getting close to a stunned human and dragging them away. After this change, this will be borderline impossible against sec (who have *almost* more flashes than Robo), and if you do it to an antag with a flash, that might just be the end of your round.

**Comment by Daundra** [comment]
Even ignoring combat potential, with this change borgs trying to avoid fights will barely be able to do so as the radius for the flash is a pretty decent chunk of screen length. For something so easy to mass produce, spam the stunning function of, and kill the borg during this time, there is really nothing a borg can do now to run away from someone chasing them with a flash. It's going to become 10x easier than it already is to subvert and kill borgs, and something that hasn't been mentioned yet, now there is a definite hard counter for malf AI borging factories, just run in, press Z, and literally an entire small room filled with borgs is stunned. It's reusable instant flashbang tier with absolutely zero drawbacks for humans. Even a person with no eye protection can spam this surrounded by borgs and be just fine while proceeding to massacre every single one.

**Comment by Grazyn** [comment]
a much needed change, borgs need to be put back in their place

**Comment by alexkar598** [comment]
I'm sure that they don't belong in hell,being a Borg is hard and the reward isint that big

On 2018-09-07, at 6:42 AM, "Grazyn" <notifications@github.com<mailto:notifications@github.com>> wrote:


a much needed change, borgs need to be put back in their place

**Comment by Hatfish** [comment]
why would anyone ever want to play a borg if they're just a worse carbon?  the way things are going people who get forceborged will just end up logging off like people who are put in perma, none of these changes are fun for anyone but powergamers who want to control every aspect of the game so they can decide who can and can't have fun.

**Comment by ExcessiveUseOfCobblestone** [comment]
Probably to roleplay.

**Review by vuonojenmustaturska — CHANGES_REQUESTED** [review]

**Comment by AlexTheSysop42** [comment]
>I say give all borgs a module to aoe stun all humans on range for 20 seconds

make it knock them out instead and make flash protection not exist

**Review by ShizCalev — CHANGES_REQUESTED** [review]
Adjacent or bust imo. this seems really shit otherwise.

**Comment by Kagus** [comment]
Borgs are _supposed_ to be terrifying, they're the Faustian bargain of the future. An incredibly useful machine-being that by its very core is bound to follow your commands...

Unless of course someone somehow changes the rules, in which case your perfect little servant gets infected with Banzaiii Buddy 3.7 and proceeds to install its hardware in your upgrade slot without any lubricant. It's the tempting illusion of control over another crewmember, with the risk that it becomes extremely fucking not under control.

But that's fine, I'm sure they'll still be cool when they're one level up from cleanbots in terms of threat. But at least give them adorable little maid outfits so that it becomes thematically appropriate to fuck them in the ass.

**Comment by Nabski89** [comment]
image
@Kagus How's this?

**Comment by Wilchenx** [comment]
@Nabski89 needs to be prettier

**Comment by CosmicScientist** [comment]
You're going to have to bash the sillybot harder to beat my downvote record.

**Comment by Mickyan** [comment]
just nerf the range of the AoE it's ridiculous that you can flash an entire room anyway, silicons or not

**Comment by zxaber** [comment]
To be honest, even if it was limited to adjacent tiles, it would still be a bit ridiculous that such a readily available item can chain-stun certain players without even needing to aim.

**Comment by wesoda25** [comment]
Just make traitor cyborgs able to enable an aoe flash resist feature. Similarly, give malf some ability to stop AOE flashes. Also, they should only affect a 5x5 area.

**Comment by AdamElTablawy** [comment]
An AoE easily spammable stun is kind of ridiculous. It turns cyborgs from 'dangerous if not adequately prepared but otherwise easily killable' into a complete joke, and turns emagged cyborgs from 'dangerous period' into another complete joke.

Cyborgs don't have ranged stuns if they're not emagged, and so by nature a cyborg has to get in flash range of you to hurt you. That alone is already a downside-without a flash, you might be screwed if you don't have eye protection, but with one, you should be safe for the most part. 

But with this you can literally just stun a borg from three tiles away by pressing Z, no aim required, and flashes are very common and very spamamble. Borgs would get absolutely destroyed.

Just make it temporarily blind the damn things if you insist on it rather than stunning them.

**Comment by AdamElTablawy** [comment]
Also, RIP Syndicate cyborgs with this.

File: code/modules/assembly/flash.dm [hunk 1]
```diff
@@ -141,18 +143,19 @@
...
 	else if(issilicon(M))
-		var/mob/living/silicon/robot/R = M
-		log_combat(user, R, "flashed", src)
-		update_icon(1)
-		R.Knockdown(rand(80,120))
-		var/diff = 5 * CONFUSION_STACK_MAX_MULTIPLIER - M.confused
-		R.confused += min(5, diff)
-		R.flash_act(affect_silicon = 1)
-		user.visible_message("<span class='disarm'>[user] overloads [R]'s sensors with the flash!</span>", "<span class='danger'>You overload [R]'s sensors with the flash!</span>")
+		borg_flash(M, user)
```
[the remark this answers is not in the document]
Comment (PKPenguin321) [reply code/modules/assembly/flash.dm]: Done at line 151

**Comment by ExcessiveUseOfCobblestone** [comment]
I think a big factor in why we don't assume the dystopian "ai rules all" thing is because the AI isn't an AI. It's a person pretending to be an AI with human flaws and traits. That has to be considered in it's balance.

**Comment by CosmicScientist** [comment]
I hope this gets in. I always thought I was missing with the AoE flash and was too worried to spam it considering how you can't easily replace burnt out ones.

>To be honest, even if it was limited to adjacent tiles, it would still be a bit ridiculous that such a readily available item could chain-stun certain players without even needing to aim.

Click sprite simulator is not fun. It's manageable against spessmen because damage = slowdown. Borgs on the other hand.

>An AoE easily spammable stun is kind of ridiculous. It turns cyborgs from 'dangerous if not adequately prepared but otherwise easily killable' into a complete joke, and turns emagged cyborgs from 'dangerous period' into another complete joke.

I feel I need to go and find my printscreens of a round where one lizard traitor rode an emagged engiborg as the borg stun armed and cremated the entire crew, sequentially, and then an inspector + ERT Officer, before the lizard decided to die gloriously via escape shuttle gibbing.

I imagine borgs are never going to be good until they stop ignoring 80% of the game's mechanics and aren't """balanced""" by situational hard counters.

>I think a big factor in why we don't assume the dystopian "ai rules all" thing is because the AI isn't an AI. It's a person pretending to be an AI with human flaws and traits. That has to be considered in it's balance.

I do wish that malf was vastly different from normal AI because that would fix balance complaints whilst not tainting regular play with big brother being a good guy.

**Comment by AdamElTablawy** [comment]
@CosmicScientist 

pocket area of effect stun you don't even need to time unlike a flashbang

yeah, that's too strong, no matter what argument you make where one engineering cyborg stun armed a bunch of idiots who don't know how to flash or flashbang.

**Comment by zxaber** [comment]
@CosmicScientist
>Click sprite simulator is not fun. It's manageable against spessmen because damage = slowdown. Borgs on the other hand.

This lets the human stun any borg, with the click of a single key, from three tiles away. What's the counter-play, here? Borgs are to follow their laws, but it would be impossible to rescue a human from an attacker if they have a flash.

"Click sprite simulator" gives the borg an even chance of removing the human from harm, either by coming at the attacker with their own flash, or just attempting to grab the human and run. At that point it's down to skill levels of the two players. But when a completely not-rare item can create a no-go zone of 49 tiles against the borg, then the borg has absolutely nothing it can do.

**Comment by CosmicScientist** [comment]
@AdamElTablawy 
tfw engiborg playing click-on-sprite-to-stun should have been robusted by click-on-sprite-to-stun flash that only sec and heads, if staffed, have access to.

And at the time, and unless something changed the status quo still is engiborg stun arm costs no power; is infinite.

Oh I'm sorry, I broke your argument, try coming back with a new one before you think I can't make one *when I already have one.*

**Comment by Wilchenx** [comment]
>tfw engiborg playing click-on-sprite-to-stun should have been robusted by click-on-sprite-to-stun flash that only sec and heads, if staffed, have access to.

what is robotics and ability to make 4 flashes from 1 metal and glass sheet

what is ion gun

what are emp nades

imho this pr makes emp nades and ion gun utter trash against borgs, just use a no-cooldown cheap flash to stun a borg by being near it

**Comment by CosmicScientist** [comment]
@zxaber it's not up to me on the distance it's supposed to be and I couldn't give a sod. If the attacker has a flash, OH WELL, continue to follow your laws and have some conflict that isn't *ctrl+click ha ha I have no slowdown good luck stopping me.*

I'm also pretty sure you missed something I said since you're claiming human versus borg = a duel of skill. Put simply, one can place weapons on tables, lockers and other spots, one slows down when hurt, one drops items when stunned, one dies to atmospheric changes, one dies to fire, one can lose limbs. The other ignores all this.

@Wilchenx last I checked not even roboticists stock up on flashes. The ion rifle gets spaced. EMP 'nades are a lie. And why would this PR make EMPs trash against borgs? Don't they still have RNG wire pulsing for that sweet sweet fuck you lockdown? To throw you a bone, you forgot research has ion carbines but to retract said bone, I seem to be the only security officer who prints off one and I can never remember where to get the pins.

**Comment by Wilchenx** [comment]
making a flash in robotics is 2-3 seconds of work
you can also make them in maintenance on meta and delta afaik

researching and printing a ion carbine is a lot more

**Comment by CosmicScientist** [comment]
I'm quite frankly offended and appalled that someone thinks speedy gonzales and sprite clicking is a battle of skill. Check your telecomms infrastructure privilege whilst you're at it.

**Comment by Wilchenx** [comment]
flashes aoe stunning humans when

**Comment by zxaber** [comment]
Other methods of dealing with borgs that don't involve flashes;
- Lockdown button (or just explode them)
- Changing laws
- EMP
- Exploit the fact that borgs have no real ranged options and just kite and shoot them

I refuse to believe that people are so unrobust that borgs are regularly unbeatable outside of the borg springing a surprise attack.

**Comment by AdamElTablawy** [comment]
You might be the saltiest bastard i’ve seen about borgs in recent memory

Plenty of people have flashes, and anyone who will be fighting a borg has flashes, too. 

A quick summary of your average flash carriers-

Sec

Heads

Engineers

Roboticists

With less frequency, scientists and assistants.

Yeah, no. It’s just a stupid balance decision all things considered. It makes cyborgs hilariously easy to kill-moreso than they’re already trivial outside of borg hordes at the moment, anyway.

You haven’t broken an argument. You haven’t even made an argument.

Plain and simple, ranged AoE hardstuns with no timer and no clickdelay are strong, no matter who has them. Rogue cyborgs-or cyborgs in general-would be next to helpless against almost anyone they would normally be fighting, because now they can’t even chance getting into melee range and dodging flash mcgee to land a knockdown or a flash themselves if the guy has no eye protection, because being half a screen away is an automatic lose if the guy presses z. 

It’s not dodgable, like an emp or a flashbang, it’s not something that can be avoided like a melee flash, and it’s not something that can be acted on to prevent its use, like the robotics console-it’s just bullshit, plain and simple.

**Comment by optimumtact** [comment]
anyway this was funny but I wasn't serious.

**Comment by ExcessiveUseOfCobblestone** [comment]
lol

**Comment by Wilchenx** [comment]
i scared him off, we won!

**Comment by PKPenguin321** [comment]
@optimumtact thoughts on https://github.com/tgstation/tgstation/pull/40062#issuecomment-418943529?

For context, pull request was closed because, in oranges' words:
>AOE flash stuns will wreck any group of borgs and immobilise them to be finished off

Which is not the case

**Comment by PKPenguin321** [comment]
Understandable, thanks for the fun

**Comment by CosmicScientist** [comment]
@AdamElTablawy lol, try in forever memory. Sillycones have been my most hated role from day 1 due to how much anti-antag, anti-*fun* they possess with *;traitor in dorms* being the tip of their valid boner iceberg for the 5 years I've been on tgstation.

Didn't know engineers now have flashes, do they come with the inventory space for it too or do you not build?

\>scientists and assistants carrying flashes
I didn't realise the roboticists shares with those who would sooner steal saws.

My argument was
\>lone emagged engiborg single handedly murders weaponless crew & e-gun armed adminspawn, this is bad

Your argument was
\>it's fine though because *some* of the crew can play who has better ping jousts

Dunno why you think EMPs have a delay considering implants and homebrews. Dunno why you think you can outrun someone cooking a flash bang. Dunno why you think rogue borgs are helpless against flammable squishies. **Dunno why you want borgs to be equal on flash fighting when they're already better at fighting.**

**Comment by Kagus** [comment]
"lol j/k"
--Oranges

**Comment by CosmicScientist** [comment]
And?

**Pull request closed without merging**